### PR TITLE
fix(142): no-console-log

### PR DESCRIPTION
**Pull request by mimshwright**
Removed console logs in src folder (also console.warn, etc). In the case where an error was being caught and logged, i'm allowing the error to be thrown

### CODE REVIEW

**Comment by mimshwright** [comment]
Fixes #142

**Comment by DanielRuf** [comment]
There are multiple issues with this as it does more than just "removing" the console logs. Removing should also mean deleting the affected lines. The other code should be unchanged imo.

File: src/component/app.tsx [hunk 1]
```diff
@@ -155,7 +155,7 @@ class App extends React.Component<AppProps> {
 		}
 
 		const designkitPath = PathUtils.join(appPath, 'build', 'designkit');
-		console.log(`Design kit path is: ${designkitPath}`);
+		/*console.log(`Design kit path is: ${designkitPath}`);*/
```
Comment (LKuechler) [inline src/component/app.tsx]: It would be better to remove the `console.log`s completely

File: src/lsg/patterns/link/demo.tsx [hunk 2]
```diff
@@ -3,36 +3,22 @@ import Link from './index';
 import * as React from 'react';
 import Space, { Size } from '../space';
 
-const clickHandler = (event: React.MouseEvent<HTMLElement>): void => console.log(event);
+const clickHandler = (event: React.MouseEvent<HTMLElement>): void => {}; //console.log(event);
```
Comment (LKuechler) [inline src/lsg/patterns/link/demo.tsx]: In the demo.tsx files it should be ok to have `console.log`s.
These files are only used when working on the livingstyleguide. This was described wrong in the task.

File: src/store/page/page_element.ts [hunk 3]
```diff
@@ -81,7 +81,7 @@ export class PageElement {
 		const element = new PageElement(pattern, false, parent);
 
 		if (!pattern) {
-			console.warn(`Ignoring unknown pattern "${patternPath}"`);
+			// console.warn(`Ignoring unknown pattern "${patternPath}"`);
```
Comment (LKuechler) [inline src/store/page/page_element.ts]: This one we should probably keep until we have a better logging

Comment (LKuechler) [reply src/store/page/page_element.ts]: I would say that in general the `console.warn`s should be keept and only the `console.log`s should be removed.
In all cases where there is a `console.warn` we have something unexpected what we probably also should show the user.

File: src/store/store.ts [hunk 4]
```diff
@@ -103,7 +103,7 @@ export class Store {
 				}
 			}
 		} catch (error) {
-			console.error(`Failed to open last styleguide or page: ${error}`);
+			throw new Error(`Failed to open last styleguide or page: ${error}`);
```
Comment (LKuechler) [inline src/store/store.ts]: This one should stay as a `console.error`. :)

File: src/component/app.tsx (hunk 1 above)
Comment (mimshwright) [reply src/component/app.tsx]: OK.

File: src/lsg/patterns/link/demo.tsx (hunk 2 above)
Comment (mimshwright) [reply src/lsg/patterns/link/demo.tsx]: Ok.

File: src/store/page/page_element.ts (hunk 3 above)
Comment (mimshwright) [reply src/store/page/page_element.ts]: Makes ssense.

File: src/store/store.ts (hunk 4 above)
Comment (mimshwright) [reply src/store/store.ts]: Understandable. I didn't want to leave any caught but unhandled errors where I had removed a console.warn or console.error. But I'll restore all of these.

**Comment by mimshwright** [comment]
removed commented console.logs. Restored .warn() and .errors()

File: src/electron/electron.html [hunk 5]
```diff
@@ -2,12 +2,10 @@
...
 <body>
 	<script>
-		const { ipcRenderer } = require('electron');
+	const { ipcRenderer } = require('electron');
 
-		ipcRenderer.on('message', function (event, text) {
-			console.log('auto-updater message:', text);
-		})
-	</script>
+ipcRenderer.on('message', function(event, text) {});
```
Comment (LKuechler) [inline src/electron/electron.html]: When we don't have a function that does anything this can be removed :)

Comment (mimshwright) [reply src/electron/electron.html]: I wonder if line 5 should be removed too... `const { ipcRenderer } = require('electron');`

File: src/electron/electron.html [hunk 6]
```diff
@@ -2,12 +2,8 @@
 
 <body>
 	<script>
-		const { ipcRenderer } = require('electron');
-
-		ipcRenderer.on('message', function (event, text) {
-			console.log('auto-updater message:', text);
-		})
-	</script>
+	const { ipcRenderer } = require('electron');
```
Comment (LKuechler) [inline src/electron/electron.html]: Sorry for just noticing this now but now the complete script tag can be removed

**Comment by faselbaum** [comment]
Thanks for the input. I'm going to close the PR since we'll stick to console log for a little bit longer.
In the future we might add a logging abstraction layer and switch to the no-console log rule.

**Pull request closed without merging**